packages/core/src/components/sections/CrossSellingShelf/CrossSellingShelf.tsx

```diff
@@ -1,37 +1,36 @@
 import { useMemo } from 'react'
 
-import type { ProductDetailsFragment_ProductFragment } from '@generated/graphql'
 import UIProductShelf from 'src/components/ui/ProductShelf'
 import { useInView } from 'react-intersection-observer'
+import { useProductsQuery } from 'src/sdk/product/useProductsQuery'
+import { usePDP } from 'src/sdk/overrides/PageProvider'
 import styles from '../ProductShelf/section.module.scss'
 import Section from '../Section'
 
 interface Props {
   items: number
   title: string
-  context: ProductDetailsFragment_ProductFragment
   kind: 'buy' | 'view'
 }
 
-const CrossSellingShelf = ({ items, title, context, kind }: Props) => {
+const CrossSellingShelf = ({ items: first, title, kind }: Props) => {
```

Comment:
We are using `first` because is the key/value used in the [useLocalizedVariable](https://github.com/vtex/faststore/blob/9b314d1a4bc88ea903899f2793cf2f3f0271f8b3/packages/core/src/sdk/product/useLocalizedVariables.ts#L22) hook from the [useProductsQuery](https://github.com/vtex/faststore/blob/9b314d1a4bc88ea903899f2793cf2f3f0271f8b3/packages/core/src/sdk/product/useProductsQuery.ts#L49) hook, that in turn uses the `search` query passing [this param](https://github.com/vtex/faststore/blob/9b314d1a4bc88ea903899f2793cf2f3f0271f8b3/packages/core/src/sdk/product/useProductsQuery.ts#L14).  

You are correct, this is the [Number of items that should return from the complete result list](https://v1.faststore.dev/reference/api/queries/search#:~:text=Arguments-,first,-*Int). But as it comes as `items` from CMS, I prefer to transform direct to `first`. 

Do you think does it worth to change the way it comes from CMS (and pass through the sync workflow) or even translate to `numberOfItems` and translate back to `first` before trigger the query?

Reply:
Definitely think it's worth doing that, but definitely don't think this is the place. I'll write a ticket for later.